src/postgres/tree/PostgresDatabaseTreeItem.ts

```diff
@@ -40,16 +60,107 @@ export class PostgresDatabaseTreeItem extends AzureParentTreeItem<ISubscriptionC
-        const sslString = process.env.POSTGRES_SSL;
-        const ssl: boolean = sslString === 'true';
-        const host: string = nonNullProp(this.parent.server, 'fullyQualifiedDomainName');
-        const clientConfig: ClientConfig = { user: fullUsername, password: password, ssl: ssl, host: host, port: 5432, database: this.databaseName };
-        const accountConnection = new Client(clientConfig);
-        const db = await pgStructure(accountConnection);
-        return db.schemas.map(schema => new PostgresSchemaTreeItem(this, schema));
+    public async loadMoreChildrenImpl(_clearCache: boolean): Promise<AzExtTreeItem[]> {
+        try {
+            const { username, password } = await this.getCredentials(false, true);
+
+            const sslString: string | undefined = process.env.POSTGRES_SSL;
+            const ssl: boolean = sslString === 'true';
+            const host: string = nonNullProp(this.parent.server, 'fullyQualifiedDomainName');
+            const clientConfig: ClientConfig = { user: username, password, ssl, host, port: 5432, database: this.databaseName };
+            const accountConnection: Client = new Client(clientConfig);
+            const db: Db = await pgStructure(accountConnection);
+            return db.schemas.map(schema => new PostgresSchemaTreeItem(this, schema));
+        } catch (error) {
+            const parsedError: IParsedError = parseError(error);
+
+            if (parsedError.errorType !== 'UserCancelledError') {
+                // tslint:disable-next-line: no-floating-promises
+                ext.ui.showWarningMessage(localize('couldNotConnect', 'Could not connect to "{0}": {1}', this.parent.label, parsedError.message));
+            }
+
+            return [new GenericTreeItem(this, {
+                contextValue: 'postgresCredentials',
+                label: localize('enterCredentials', 'Enter server credentials to connect to "{0}"...', this.parent.label),
+                commandId: 'cosmosDB.getPostgresCredentials'
+            })];
+        }
+    }
+
+    public async getCredentials(forcePrompt: boolean, warnBeforePrompting: boolean): Promise<{ username: string, password: string }> {
+        let username: string | undefined;
+        let password: string | undefined;
+
+        if (!forcePrompt) {
+            const storedValue: string | undefined = ext.context.globalState.get(this._serviceName);
+            if (storedValue && this._keytar) {
+                const servers: IPersistedServer[] = JSON.parse(storedValue);
+                for (const server of servers) {
+                    if (server.id === this._serverId) {
+                        username = server.username;
+                        password = await this._keytar.getPassword(this._serviceName, this._serverId) || undefined;
+                        break;
+                    }
+                }
+            }
+        }
+
+        if (!username || !password) {
+            if (warnBeforePrompting) {
```

Comment:
What's the point of warning the user before prompting them?

Reply:
Matt requested that we prompt before asking for credentials so what's happening is more obvious

Reply:
But that warning didn't make sense when you entered credentials via the cmd palette command so I removed it in that case

Reply:
I'm with Nathan on this. Seems like the prompt in the input box should have the message being shown here

Reply:
If we want a separate prompt to make it more obvious what's going on, how about we count your nice tree item as that prompt:
<img width="345" alt="Screen Shot 2020-03-27 at 11 22 53 AM" src="https://user-images.githubusercontent.com/11282622/77787753-5293c580-701d-11ea-8369-6f117a7e3c93.png">

Aka don't prompt them automatically and just make them click that. It's the same amount of clicks as your current "Continue" button 😂.